rclpy/src/rclpy/_rclpy_pybind11.cpp

```diff
@@ -62,6 +63,22 @@ PYBIND11_MODULE(_rclpy_pybind11, m) {
   py::register_exception<rclpy::UnsupportedEventTypeError>(
     m, "UnsupportedEventTypeError", rclerror.ptr());
 
+  py::class_<rclpy::QoSCheckCompatibleResult>(m, "QoSCheckCompatibleResult")
+  .def(py::init<>())
+  .def_property_readonly(
+    "compatibility", [](const rclpy::QoSCheckCompatibleResult & result) {
+      return static_cast<int>(result.compatibility);
+    })
+  .def_property_readonly(
+    "reason", [](const rclpy::QoSCheckCompatibleResult & result) {
+      return result.reason;
+    })
+  .def(
+    "__repr__", [](const rclpy::QoSCheckCompatibleResult & result) {
+      (void) result;
+      return "Result type for checking QoS compatibility with result";
```

Comment:
I think this would fit better as a class doc string than the output of `__repr__`. I would ditch the `def` of `__repr__` completely and add this string as a third parameter to the constructor.

```c++
  py::class_<rclpy::QoSCheckCompatibleResult>(
    m, "QoSCheckCompatibleResult",
    "Result type for checking QoS compatibility with result")
...
```

The result is when some one does `help(QoSCheckCompatibleResult)` they see this string describing the type.

```
class QoSCheckCompatibleResult(pybind11_builtins.pybind11_object)
 |  Result type for checking QoS compatibility with result
...
```